app/uk/gov/hmrc/gform/validation/ComponentValidator.scala

```diff
@@ -654,6 +673,169 @@ object ComponentValidator {
     }
 
+  private def validateSterling(
+    fieldValue: FormComponent,
+    value: String,
+    isPositive: Boolean,
+    isWhole: Boolean
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] = {
+    val maxWhole = ValidationValues.sterlingLength
+    val WholeShape = "([+-]?)(\\d+(,\\d{3})*?)[.]?".r
+    val FractionalShape = "([+-]?)(\\d*(,\\d{3})*?)[.](\\d+)".r
+    TextConstraint.filterNumberValue(value) match {
+      case FractionalShape(_, _, _, fractional) if !isWhole && fractional.length > 2 =>
+        nonNumericSterlingFailure(fieldValue, value)
+      case FractionalShape(_, _, _, fractional) if isWhole =>
+        wholeSterlingFailure(fieldValue, value)
+      case WholeShape(_, whole, _) if surpassMaxLength(whole, maxWhole) =>
+        maxDigitSterlingFailure(fieldValue, value, maxWhole)
+      case WholeShape("-", _, _) if isPositive         => positiveSterlingFailure(fieldValue, value)
+      case FractionalShape("-", _, _, _) if isPositive => positiveSterlingFailure(fieldValue, value)
+      case WholeShape(_, _, _)                         => validationSuccess
+      case FractionalShape(_, _, _, _) if !isWhole     => validationSuccess
+      case _                                           => nonNumericSterlingFailure(fieldValue, value)
+    }
+  }
+
+  private def nonNumericSterlingFailure(
+    fieldValue: FormComponent,
+    value: String
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] =
+    (fieldValue.errorShortName, fieldValue.errorShortNameStart, fieldValue.errorExample) match {
+      case (None, None, maybeErrorExample) =>
+        val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorPattern,
+          Some(List(SmartString.blank.trasform(_ => "an amount", _ => "swm").value(), errorExample.value()))
+        )
+
+      case (Some(errorShortName), None, maybeErrorExample) =>
+        val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorPattern,
+          Some(List(errorShortName.value(), errorExample.value()))
+        )
+      case (_, Some(errorShortNameStart), maybeErrorExample) =>
+        val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
+        validationFailure(
+          fieldValue,
+          genericSterlingErrorPatternStart,
+          Some(List(errorShortNameStart.value(), errorExample.value()))
+        )
+    }
+
+  private def maxDigitSterlingFailure(
+    fieldValue: FormComponent,
+    value: String,
+    maxWhole: Int
+  )(implicit
+    messages: Messages,
+    sse: SmartStringEvaluator
+  ): ValidatedType[Unit] =
+    (fieldValue.errorShortNameStart, fieldValue.errorExample) match {
+      case (None, maybeErrorExample) =>
+        val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
```

Comment:
```
val errorExample = maybeErrorExample.getOrElse(SmartString.blank)
```
This is duplicated 9 times, can we do something to reduce it to have this logic only once?